File: MonitoringServiceAPI/MonitoringServiceAPI/Models/PushServerSetting.cs
using System.ComponentModel.DataAnnotations;

namespace MonitoringServiceAPI.Models
{
    public class PushServerSetting
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public bool IsEnabled { get; set; } = false;

        [Required]
        [StringLength(500)]
        public string DataFolderPath { get; set; } = string.Empty;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [StringLength(100)]
        public string? UpdatedBy { get; set; }
    }
}

File: AzureGateway.UI/src/models/PushServerSetting.ts
export interface PushServerSetting {
    id: number;
    isEnabled: boolean;
    dataFolderPath: string;
    updatedAt: string;
    updatedBy: string | null;
}

export interface UpdatePushServerSettingRequest {
    isEnabled: boolean;
    dataFolderPath: string;
    updatedBy?: string;
}

export interface EnableServiceRequest {
    updatedBy?: string;
}

export interface DisableServiceRequest {
    updatedBy?: string;
}

